cni-plugin/pkg/types/types.go

```diff
@@ -123,6 +123,11 @@ type NetConf struct {
 	// Options below here are deprecated.
 	EtcdAuthority string `json:"etcd_authority"`
 	Hostname      string `json:"hostname"`
+
+	// If WaitForReadyEndpoint = true then cni waits until
+	// readiness endpoint(ReadinessEndpoint) is ready.
+	WaitForReadyEndpoint bool   `json:"wait_for_ready_endpoint"`
```

Comment:
nit: can we rename this to `WaitForEndpointReady`/`wait_for_endpoint_ready`?

Reply:
Do we actually need two knobs here?

I'm thinking we just have `ReadinessEndpoint` and if it is specified, then we perform the check, and if it is not specified we don't perform the check. No need for a second on/off switch.

On top of that, I think we might want this to be `ReadinessEndpoints []string` - because Calico exposes two different HTTP readiness endpoints, and there may be other systems that we want to plug Calico into to gate launching of pods down the line.

Reply:
Made these modifications.